Sources/NIOExtras/LineBasedFrameDecoder.swift

```diff
@@ -0,0 +1,86 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+///
+/// A decoder that splits incoming `ByteBuffer`s around line end
+/// character(s) (`'\n'` or `'\r\n'`).
+///
+/// Let's, for example, consider the following received buffer:
+///
+///     +----+-------+------------+
+///     | AB | C\nDE | F\r\nGHI\n |
```

Comment:
should we make it configurable what sequences work as splitting sequence?

Reply:
@weissi I started down this path but thought we can do a few optimisations knowing the delimiter is a single character (`\n`). Happy to generalise this decoder if you feel this would be more useful. Though I suspect `\n` or `\r\n` will represent the majority of use cases